frontend/packages/schema-editor/src/components/SchemaTree/SchemaNode/ActionButtons/ActionButtons.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useCallback } from 'react';
+import { ReferenceIcon, TrashIcon } from '@studio/icons';
+import { useTranslation } from 'react-i18next';
+import classes from './ActionButtons.module.css';
+import cn from 'classnames';
+import { isNodeValidParent, isReference } from '../../../../../../schema-model';
+import { ActionButton } from './ActionButton';
+import { AddPropertyMenu } from './AddPropertyMenu';
+import { useSavableSchemaModel } from '@altinn/schema-editor/hooks/useSavableSchemaModel';
+import { SavableSchemaModel } from '@altinn/schema-editor/classes/SavableSchemaModel';
+
+export interface ActionButtonsProps {
+  pointer: string;
+  className: string;
+}
+
+export const ActionButtons = ({ pointer, className }: ActionButtonsProps) => {
+  const savableModel = useSavableSchemaModel();
+  const deleteNode = useDeleteNode(pointer, savableModel);
+  const node = savableModel.getNode(pointer);
+
+  const convertToReference = () => savableModel.convertToDefinition(pointer);
+
+  return (
+    <div className={cn(classes.root, className)}>
+      {isNodeValidParent(node) && <AddPropertyMenu pointer={pointer} />}
+      {!isReference(node) && (
+        <ActionButton
+          icon={<ReferenceIcon />}
+          titleKey='schema_editor.promote'
+          onClick={convertToReference}
+        />
+      )}
+      <ActionButton
+        color='danger'
+        icon={<TrashIcon />}
+        titleKey='general.delete'
+        onClick={deleteNode}
+      />
+    </div>
+  );
+};
+
+const useDeleteNode = (pointer: string, savableModel: SavableSchemaModel) => {
```

Comment:
I decided to keep it inside the file since it is only used internally in this file. Moving it to a separate file will expose it to the rest of the system although it is not meant to be accessed from anywhere else.

Reply:
That's fair enough 😄